internal/comby/comby.go

```diff
@@ -0,0 +1,100 @@
+package comby
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+const combyPath = "comby"
+const numWorkers = 8
+
+func exists() error {
+	_, err := exec.LookPath(combyPath)
+	if err != nil {
+		return errors.New("comby is not installed on the PATH. Try running 'bash <(curl -sL get.comby.dev)'")
+	}
+	return nil
+}
+
+func rawArgs(args Args) (rawArgs []string) {
+	rawArgs = append(rawArgs, args.MatchTemplate, args.RewriteTemplate)
+	rawArgs = append(rawArgs, args.FilePatterns...)
+	rawArgs = append(rawArgs, "-json-lines")
+
+	if args.MatchOnly {
+		rawArgs = append(rawArgs, "-match-only")
+	} else {
+		rawArgs = append(rawArgs, "-json-only-diff")
+	}
+
+	rawArgs = append(rawArgs, "-jobs", strconv.Itoa(numWorkers))
+
+	if args.Matcher != "" {
+		rawArgs = append(rawArgs, "-matcher", args.Matcher)
+	}
+
+	switch i := args.Input.(type) {
+	case ZipPath:
+		rawArgs = append(rawArgs, "-zip", string(i))
+	case DirPath:
+		rawArgs = append(rawArgs, "-directory", string(i))
+	default:
+		panic("unreachable")
```

Comment:
Haha, just commented on this, because `DirPath` looks like should corresponding to "-directory".

---

For the panic message, what do you think of this? It should never happen, but in case it happens, could save some debug efforts:

```suggestion
		panic(fmt.Sprintf("unrecognized input type: %T", i))
```

Reply:
As far as I know, this is the [idiomatic way](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/golang/go%24+panic%28%22unreachable%22%29&patternType=literal) to express exhaustive pattern matching in Go. The "unreachable" error message implies that this shouldn't ever be reachable, whereas the informative message for debugging doesn't communicate the same intent or severity, I think.

Reply:
> As far as I know, this is the [idiomatic way](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/golang/go%24+panic%28%22unreachable%22%29&patternType=literal) to express exhaustive pattern matching in Go.

TIL 👍 

I also see there is often a line of `FatalF(...)` right before the `panic(...)`. Which serves the same purpose I was hoping for.

![image](https://user-images.githubusercontent.com/2946214/67811897-07b85680-fa5b-11e9-9f80-2248c21fa01f.png)

Reply:
Oh, good find. So I'll add a `Fatalf` and we can have both :)

Reply:
`Fatalf` is an internal Go compiler thing; I ended up just using log15.